src/AmqpCompat/Driver/Amqplib/Processor/ValueProcessor.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * PHP AMQP-Compat - php-amqp/ext-amqp compatibility.
+ * Copyright (c) Dan Phillimore (asmblah)
+ * https://github.com/asmblah/php-amqp-compat/
+ *
+ * Released under the MIT license.
+ * https://github.com/asmblah/php-amqp-compat/raw/main/MIT-LICENSE.txt
+ */
+
+declare(strict_types=1);
+
+namespace Asmblah\PhpAmqpCompat\Driver\Amqplib\Processor;
+
+use AMQPTimestamp;
+use Asmblah\PhpAmqpCompat\Driver\Common\Processor\ValueProcessorInterface;
+use DateTime;
+use DateTimeInterface;
+
+/**
+ * Class ValueProcessor.
+ *
+ * Transforms AMQP values either originating from or destined for the driver.
+ *
+ * @author Dan Phillimore <dan@ovms.co>
+ */
+class ValueProcessor implements ValueProcessorInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function processValueForDriver(mixed $value): mixed
+    {
+        if (is_array($value)) {
+            foreach ($value as $key => $elementValue) {
+                $value[$key] = $this->processValueForDriver($elementValue);
+            }
+
+            return $value;
+        }
+
+        // TODO: Handle AMQPDecimal.
```

Comment:
Leftover TODO here

Reply:
No, this is a future TODO, which I'll link with a GitHub issue 👍

Reply:
Change of tack, this is now implemented in https://github.com/asmblah/php-amqp-compat/pull/7 😄